modules/core/src/main/java/org/apache/synapse/deployers/TemplateDeployer.java

```diff
@@ -214,22 +215,22 @@ public void undeploySynapseArtifact(String artifactName) {
 
         try {
             Template st = null;
-            try {
-                st = getSynapseConfiguration().getEndpointTemplate(artifactName);
-            } catch (SynapseException e) {
-                //could not locate an endpoint template for this particular un-delpoyment. This name refers
-                //probably to a sequence tempalte. Thus if  throws a Synapse exception with following message
-                //catch n log that and continue this process for undeployment of a sequence template
-                if (e.getMessage().indexOf("Cannot locate an either local or remote entry for key") != -1) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Undeploying template is not of endpoint type. Undeployer will now check " +
-                                  "for sequence template for the key: " + artifactName);
-                    }
-                } else {
-                    //different error hence stop undeployment/report failure
+            Map<String, Template> templates = getSynapseConfiguration().getEndpointTemplates();
+            if (templates.containsKey(artifactName)) {
+                try {
+                    st = getSynapseConfiguration().getEndpointTemplate(artifactName);
+                } catch (SynapseException e) {
```

Comment:
Why are we catching the exception and just throwing it?

Reply:
Actually yes. Because there is nothing we should do except stop the undeployment flow.